sycl/doc/extensions/proposed/sycl_ext_oneapi_append_and_shift.asciidoc

```diff
@@ -0,0 +1,183 @@
+
+Similarly, the proposed extension of `sycl::shift_group_right` returns `x` of the work-item with group local id `id-delta`, if `id-delta` is greater than or equal to 0 and `to_prepend` of the work-item with group local id `sg_size+(id-delta)`, otherwise.
+
+While this feature can easily be reproduced manually, as indicated below in the section "possible implementation", an extension is feasible since the underlying `pass:[__]spirv_SubgroupShuffleDownINTEL` and `pass:[__]spirv_SubgroupShuffleUpINTEL` already take the proposed two values and support the required capabilities.
+
+=== Example
+Assuming the group `g` consists of 4 work items (`sg_size = 4`), which are enumerated `0-3` (`WI0`, `WI1`, `WI2`, `WI3`).
+Each work item holds a value `x` and a value `to_append`. To indicate the relation between a work item and its value, we enumerate the values as `x0`, `x1`, `x2`, `x3` for the values associated with work items `WI0`, `WI1`, `WI2`, `WI3`, respectively. Similarly for `to_append0`, etc.
+
+Assuming `append_and_shift_group_left` is called with `delta=2`, we would get the following return values:
+[%header,cols="1,1,1,1,1"]
+|===
+|Work Item
+|WI0
+|WI1
+|WI2
+|WI3
+
+|Input x
+|x0
+|x1
+|x2
+|x3
+	
+|Input to_append
+|to_append0
+|to_append1
+|to_append2
+|to_append3
+
+|Returned Value
+|x2
+|x3
+|to_append0
+|to_append1
+|===
+
+Similarly, calling `prepend_and_shift_group_right` (with enumerated `to_prepend0` to `to_prepend3` analogously to the above `to_append`) with `delta=3` would yield
+
+[%header,cols="1,1,1,1,1"]
+|===
+|Work Item
+|WI0
+|WI1
+|WI2
+|WI3
+
+|Input x
+|x0
+|x1
+|x2
+|x3
+	
+|Input to_prepend
+|to_prepend0
+|to_prepend1
+|to_prepend2
+|to_prepend3
+
+|Returned Value
+|to_prepend1
+|to_prepend2
+|to_prepend3
+|x0
+|===
+
+
+
+== Specification
+
+=== Feature test macro
+This extension provides a feature-test macro as described in the core SYCL specification. An implementation supporting this extension must predefine the macro `SYCL_EXT_ONEAPI_APPEND_AND_SHIFT` to one of the values defined in the table below. Applications can test for the existence of this macro to determine if the implementation supports this feature, or applications can test the macro's value to determine which of the extension's features the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+	
+|1
+|The APIs of this experimental extension are not versioned, so the feature-test macro always has this value.
+|===
+
+=== API
+
+```c++
+template <typename Group, typename T>
+T sycl::ext::oneapi::append_and_shift_group_left(Group g, T x, T to_append, Group::linear_id_type delta=1);
+```

```

Comment:
Kind of a nit, but our standard way of showing the synopsis is to show the declaration (not definition), which is like this:

```
namespace sycl::ext::oneapi {

template <typename Group, typename T>
T append_and_shift_group_left(Group g, T x, T to_append, Group::linear_id_type delta=1)

} // namespace sycl::ext::oneapi
```

Reply:
I updated it.